logging/src/main/java/org/jboss/as/logging/formatters/StructuredFormatterResourceDefinition.java

```diff
@@ -0,0 +1,462 @@
+            META_DATA,
+            PRETTY_PRINT,
+            PRINT_DETAILS,
+            RECORD_DELIMITER,
+            ZONE_ID,
+    };
+
+    private static final OperationStepHandler WRITE = new LoggingOperations.LoggingWriteAttributeHandler(ATTRIBUTES) {
+
+        @SuppressWarnings("OverlyStrongTypeCast")
+        @Override
+        protected boolean applyUpdate(final OperationContext context, final String attributeName, final String addressName,
+                                      final ModelNode value, final LogContextConfiguration logContextConfiguration) throws OperationFailedException {
+            final FormatterConfiguration configuration = logContextConfiguration.getFormatterConfiguration(addressName);
+            if (attributeName.equals(META_DATA.getName())) {
+                final String metaData = modelValueToMetaData(value);
+                if (metaData != null) {
+                    configuration.setPropertyValueString("metaData", metaData);
+                } else {
+                    configuration.removeProperty("metaData");
+                }
+            } else if (attributeName.equals(KEY_OVERRIDES.getName())) {
+                // Require a restart of the resource
+                return true;
+            } else {
+                for (AttributeDefinition attribute : ATTRIBUTES) {
+                    if (attribute.getName().equals(attributeName)) {
+                        if (attribute instanceof PropertyAttributeDefinition) {
+                            final PropertyAttributeDefinition propertyAttribute = (PropertyAttributeDefinition) attribute;
+                            if (value.isDefined()) {
+                                final ModelNodeResolver<String> resolver = propertyAttribute.resolver();
+                                String resolvedValue = value.asString();
+                                if (resolver != null) {
+                                    resolvedValue = resolver.resolveValue(context, value);
+                                }
+                                configuration.setPropertyValueString(propertyAttribute.getPropertyName(), resolvedValue);
+                            } else {
+                                configuration.removeProperty(propertyAttribute.getPropertyName());
+                            }
+                        } else {
+                            if (value.isDefined()) {
+                                configuration.setPropertyValueString(attributeName, value.asString());
+                            } else {
+                                configuration.removeProperty(attributeName);
+                            }
+                        }
+                        break;
+                    }
+                }
+            }
+            return false;
+        }
+    };
+
+    /**
+     * A step handler to remove
+     */
+    private static final OperationStepHandler REMOVE = new LoggingOperations.LoggingRemoveOperationStepHandler() {
```

Comment:
Why isn't LoggingOperations.LoggingRemoveOperationStepHandler inheriting from AbstractRemoveStepHandler so that you get most of the clean up done by the super class?

It was my understanding that AbstractRemoveStepHandler and AbstractAddStepHandler where the "correct" base classes to define :add and :remove attributes and make sure that the operations were behaving as expected (wrt to cleaning up resources, rolling back ops, etc.)

Reply:
At some point this is something I would like to fix. However currently logging has some special cases that need to be handled. The main one is when a change is made the `logging.properties` file needs to be rewritten. Hopefully soon we can remove the `logging.properties` which will trigger a rewrite of a lot of these :)